pkg/controller/registry/registry_client.go

```diff
@@ -0,0 +1,112 @@
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -o resolver/fakes/fake_registry_client_interface.go . RegistryClientInterface
+package registry
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	registryapi "github.com/operator-framework/operator-registry/pkg/api"
+	"github.com/operator-framework/operator-registry/pkg/client"
+	opregistry "github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+type ChannelEntryStream interface {
+	Recv() (*registryapi.ChannelEntry, error)
+}
+
+type RegistryClientInterface interface {
+	client.Interface
+	FindBundleThatProvides(ctx context.Context, group, version, kind, pkgName string) (*registryapi.Bundle, error)
+	GetLatestChannelEntriesThatProvide(ctx context.Context, group, version, kind string) (*ChannelEntryIterator, error)
+}
+
+type ChannelEntryIterator struct {
+	stream ChannelEntryStream
+	error  error
+}
+
+func NewChannelEntryIterator(stream ChannelEntryStream) *ChannelEntryIterator {
+	return &ChannelEntryIterator{stream: stream}
+}
+
+func (ceit *ChannelEntryIterator) Next() *registryapi.ChannelEntry {
+	if ceit.error != nil {
+		return nil
+	}
+	next, err := ceit.stream.Recv()
+	if err == io.EOF {
+		return nil
+	}
+	if err != nil {
+		ceit.error = err
+	}
+	return next
+}
+
+func (ceit *ChannelEntryIterator) Error() error {
+	return ceit.error
+}
+
+type RegistryClient struct {
+	client.Interface
+	Client *client.Client
+}
+
+func NewRegistryClient(client *client.Client) *RegistryClient {
+	return &RegistryClient{client, client}
+}
+
+var _ RegistryClientInterface = &RegistryClient{}
+var _ client.Interface = &RegistryClient{}
```

Comment:
I added this to the example for show. The line above covers this check.

Reply:
ack